### ISSUE TITLE
Can't login anymore

### ISSUE THREAD

**Issue by drehtisch** [lloydtorres/stately#16]
I recently changed my password. And it works fine on the website. But if i try to login with that same combination on the app it says "invalid nation name or password". 

I tried to delete the app data and to reinstall the app but the problem persists. Does someone have an idea how to solve that problem?

I'm using KeepassForAndroid could this be an problem?

**Comment by lloydtorres** [comment]
Hi! This is a major issue that's been plaguing a fair amount of users. I haven't been able to track it down to something specific since I haven't been able to reproduce it.

I have some ideas on how to possibly fix it or at least mitigate it, but I'll need someone facing the issue to test out my fixes. Would you mind if I sent you beta versions of the app to test out?

**Comment by drehtisch** [comment]
I do have this issue on different devices as well. I wouldn't mind to help solving this Issue by testing an beta version.

**Comment by drehtisch** [comment]
Any updates on this? The issue still persists :(

**Comment by Tomaasci** [comment]
Yes, it still persists.

**Comment by SJTheGreat06** [comment]
Yep, same problem with me as well! I can't login in on the app...

**Comment by iSaluki** [comment]
I've just observed this issue on my first install of the app, making it unusable for me.

**Comment by lloydtorres** [comment]
Sorry all for the continued login issues. Unfortunately, I'm not able to repro the issue on my end.

To help debug the issue, can folks comment on:
* Which Android device and OS version are they using
* Which keyboard is used to type on the device (e.g. default system keyboard, a keyboard app, physical keyboard)
* Whether copy-pasting credentials (either manually or through a password manager) resolves the issue
* Whether resetting nation passwords on NationStates resolves the issue

**Comment by Erz3** [comment]
could it be produced by specials characters on passwords? I have got the same problem, I would check tomorrow with and easy password.

**Comment by Erz3** [comment]
> could it be produced by specials characters on passwords? I have got the same problem, I would check tomorrow with and easy password.

Checked, there are problems with special characters. For example, I have not problems with "@" and "*", but impossible with "&".

**Comment by lloydtorres** [comment]
@Erz3 Thanks so much for testing this! Part of this could definitely be an encoding issue, as you've called out with `&`. However, from previous user reports, I do recall simple alphanumeric passwords also being impacted somehow.

The bit of the app which sends the password to NationStates during login is here: https://github.com/lloydtorres/stately/blob/master/Stately/app/src/main/java/com/lloydtorres/stately/helpers/network/NSStringRequest.java#L125-L128

I'll play around with changing the encoding here once I have some free time.

**Comment by samolukadjo** [comment]
I don't encounter this bug only when using a letter only password

**Comment by BillyMcGuyer** [comment]
Just want to add that - for whatever reason - @samolukadjo fixed the problem for me. Changing my password a couple of times gave no results. However, removing the numbers from my password (leaving it consisting of letters and symbols still) randomly decided to let me log in again.

**Comment by BillyMcGuyer** [comment]
> Just want to add that - for whatever reason - @samolukadjo fixed the problem for me. Changing my password a couple of times gave no results. However, removing the numbers from my password (leaving it consisting of letters and symbols still) randomly decided to let me log in again.

I was going to edit, but I'm not remotely familiar with how GitHub works so idk if it would notify of an edited message.

Turns out I was wrong, the numbers were not the issue for me. For me it seems the app doesn't like any password longer than 12 characters - I made one with 12 characters (mixed letters and symbols) and it worked, bumping it up to 16 broke it again. I knocked it back down to 13 and still didn't work.

A password of 12 characters - still containing letters, symbols **_and_** numbers - worked to let me in again.
